### PR TITLE
MGMT-15177: Align feature support level to support platform as filterable feature

### PR DESCRIPTION
**Pull request by eliorerz**
This PR was created to allow the UI to stop using `user-managed-networking` parameter and to use only the `platform` parameter instead.

Changes in this PR:
- Add the ability to filter feature support by platform type
- Add baremetal and none platforms as valid platforms
- Add a "new feature" that indicates whether each platform's network is managed or not - called "Platform Managed Networking",  this feature cannot be set as is driven by the platform type
  - None and OCI - `Platform Managed Networking` enabled
  - Baremetal, vSphere and Nutanix - `Platform Managed Networking` disabled

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

/cc @eliorerz

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -4070,6 +4075,8 @@ definitions:
       - 'CNV'
       - 'MCE'
       - 'NUTANIX_INTEGRATION'
+      - 'BAREMETAL_INTEGRATION'
```
[the remark this answers is not in the document]
Comment (eliorerz) [reply swagger.yaml:4078]: I wanted to keep the convention, do you think we should rename nutanix and vsphere?

**Comment by eliorerz** [comment]
/cc @avishayt
PR is ready fore review

**Review by avishayt — APPROVED** [review]

**Comment by eliorerz** [comment]
/override ci/prow/edge-e2e-ai-operator-ztp-hypershift-zero-nodes

**Pull request merged**